### PR TITLE
Change the openjdk to adoptjdk

### PR DESCRIPTION
**Pull request by MaSven**
I think people will get confused with the openjdk suffic since openjdk releases it's own builds. So i changed the suffix for the adoptopenjdk to -adoptopenjdk. Also to make clear that this are not the openjdk builds.

### CODE REVIEW

**Comment by marc0der** [comment]
Not certain I agree. AdoptOpenJDK are actually the official openjdk builds now, and the ones that you refer to are merely what debian/fedora etc package themselves. I'd prefer rolling with openjdk, because this is precisely what it is in it's purest form.

**Comment by MaSven** [comment]
ok so i was confused because here http://jdk.java.net/10/ java 10 is build for all major platforms but in adoptopenjdk https://adoptopenjdk.net/releases.html?variant=openjdk10 only linux and mac is released.

File: src/main/scala/io/sdkman/changelogs/Migrations.scala [hunk 1]
```diff
@@ -35,16 +35,32 @@ class Migrations {
...
     updateCandidateDefault("scala", "2.12.5")
+
+  }
+
+  @ChangeSet(order = "009", id = "009-add_java_10_openjdk", author = "MaSven")
+  def migrate009(implicit db: MongoDatabase) = {
+    insertVersions(
+      CandidateVersion("java", "10-openjdk", Some("WINDOWS_64"), "https://download.java.net/java/GA/jdk10/10/binaries/openjdk-10_windows-x64_bin.tar.gz"),
+      CandidateVersion("java", "10-openjdk", Some("MAC_OSX"), "https://download.java.net/java/GA/jdk10/10/binaries/openjdk-10_osx-x64_bin.tar.gz"),
+      CandidateVersion("java", "10-openjdk", Some("LINUX_64"), "https://download.java.net/java/GA/jdk10/10/binaries/openjdk-10_linux-x64_bin.tar.gz"))
```
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/Migrations.scala]: These are actually Oracle's openjdk distributions. I think adopt are considered more official by now.

File: src/main/scala/io/sdkman/changelogs/Migrations.scala [hunk 2]
```diff
@@ -35,16 +35,32 @@ class Migrations {
...
-
   @ChangeSet(order = "006", id = "003-add_maven_353", author = "hho")
   def migrate006(implicit db: MongoDatabase) = {
     insertVersions(CandidateVersion("maven", "3.5.3", Some("UNIVERSAL"), "https://archive.apache.org/dist/maven/maven-3/3.5.3/binaries/apache-maven-3.5.3-bin.zip"))
     updateCandidateDefault("maven", "3.5.3")
   }
 
+  @ChangeSet(order = "008", id = "008-change_openjdk_to_adoptopenjdk", author = "MaSven")
+  def migrate008(implicit db: MongoDatabase) = {
+    updateVersion("9u181-openjdk", "9u181-adoptopenjdk")
```
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/Migrations.scala]: This name is too verbose, and takes up too much real estate in the list view. I think we should stick with openjdk for now.

File: src/main/scala/io/sdkman/changelogs/Migrations.scala [hunk 3]
```diff
@@ -1,6 +1,6 @@
 package io.sdkman.changelogs
 
-import com.github.mongobee.changeset.{ChangeLog, ChangeSet}
+import com.github.mongobee.changeset.{ ChangeLog, ChangeSet }
```
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/Migrations.scala]: Not related to the PR, and not the standard scala code style.

File: src/main/scala/io/sdkman/changelogs/package.scala [hunk 4]
```diff
@@ -1,26 +1,29 @@
...
 
-  case class CandidateVersion(candidate: String,
-                              version: String,
-                              platform: Option[String],
-                              url: String)
+  case class CandidateVersion(
+    candidate: String,
+    version:   String,
+    platform:  Option[String],
+    url:       String)
```
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/package.scala]: Reformatting not a part of this PR. Also, once again not standard scala code style.

File: src/main/scala/io/sdkman/changelogs/package.scala [hunk 5]
```diff
@@ -44,4 +47,9 @@ package object changelogs {
 
   def updateCandidateDefault(candidate: String, version: String)(implicit db: MongoDatabase): Document =
     db.getCollection("candidates").findOneAndUpdate(Filters.eq("candidate", candidate), Updates.set("default", version))
+
+  def updateVersion(version: String, newVersion: String)(implicit db: MongoDatabase): Unit = {
+    db.getCollection("versions").updateMany(Filters.eq("version", version), Updates.set("version", newVersion))
```
Comment (marc0der) [inline src/main/scala/io/sdkman/changelogs/package.scala]: `updateMany` could potentially be dangerous as version could be very ambiguous. I like the idea of adding this helper function though...

**Comment by marc0der** [comment]
@MaSven It's late here now, let me sleep on this a bit. I think adoptjdk does sound kinda cool ;-)

File: src/main/scala/io/sdkman/changelogs/Migrations.scala (hunk 3 above)
Comment (MaSven) [reply src/main/scala/io/sdkman/changelogs/Migrations.scala]: Sorry not a scala dev and just downloaded the scala ide for the pull requests. Do you have some form of setting for the scala ide so i have the right formatting?

File: src/main/scala/io/sdkman/changelogs/package.scala (hunk 5 above)
Comment (MaSven) [reply src/main/scala/io/sdkman/changelogs/package.scala]: Ok so i think a better way would be to find them one by one and after that update the version?

**Comment by MaSven** [comment]
yes and it is shorter. Makes less typing.

File: src/main/scala/io/sdkman/changelogs/Migrations.scala (hunk 1 above)
Comment (MaSven) [reply src/main/scala/io/sdkman/changelogs/Migrations.scala]: This wasn't intended for this PR. I pushed to the wrong branch here sorry for that. I wanted to make a new PR for this one.

**Comment by marc0der** [comment]
After the [Twitter poll](https://twitter.com/sdkman_/status/977076310571978757) on Friday it is conclusive that everyone _wants_ the openjdk from java.net. This will become our official openjdk distribution of choice and all others will be removed soon (no point in hosting multiple openjdk distros as they all contain the same). As a result of this, I'm closing this PR for now.

**Comment by tony--** [comment]
Any chance of revisiting this, @marc0der @MaSven ?
I plan to use adoptopenjdk because of the longer support life.

**Pull request closed without merging**